force-app/main/default/classes/IBMDiscoveryV1.cls

```diff
@@ -532,8 +536,11 @@ public class IBMDiscoveryV1 extends IBMWatsonService {
   public IBMDiscoveryV1Models.QueryResponse federatedQuery(IBMDiscoveryV1Models.FederatedQueryOptions federatedQueryOptions) {
     IBMWatsonValidator.notNull(federatedQueryOptions, 'federatedQueryOptions cannot be null');
     IBMWatsonRequestBuilder builder = IBMWatsonRequestBuilder.httpGet(getEndPoint() + String.format('/v1/environments/{0}/query', new String[]{ federatedQueryOptions.environmentId() }));
+    if(federatedQueryOptions.summary() != null){
+      builder = IBMWatsonRequestBuilder.httpGet(getEndPoint() + String.format('/v1/environments/{0}/collections/{1}/query', new String[]{ federatedQueryOptions.environmentId(), federatedQueryOptions.collectionIds().get(0) }));
```

Comment:
you need to use the `QueryOptions` not `FederatedQueryOptions`

Reply:
ok. I will update